### DISCUSSION TITLE
Environment in matrix substitution not work as expected

### DISCUSSION THREAD

**Discussion by liudonghua123** [community/community#26151] [Actions]
<p>I have a strategy configurations like the following.</p>
<pre><code>strategy:
      matrix:
        os: [ubuntu-latest, windows-latest, macOS-latest]
        include:
          - os: ubuntu-latest
            output-name: ${GITHUB_REPOSITORY##*/}-linux
          - os: macOS-latest
            output-name: ${GITHUB_REPOSITORY##*/}-mac
          - os: windows-latest
            output-name: ${GITHUB_REPOSITORY##*/}-windows.exe
</code></pre>
<p>However in the following steps where I use ${{ matrix.output-name }} in windows-latest environment, I got the incorrect -windows.exe.</p>

**Comment by Yanjingzhu** [comment]
<p>How did you use ${{ matrix.output-name }} in the next steps? If it is a run step, in windows it will use powershell as the default shell. You could specify shell to use bash. </p>
<pre><code>- name: matrix
      run: |
          echo ${{ matrix.output-name }}
      shell: bash
</code></pre>

**Comment by liudonghua123** [comment, marked as answer]
<p>I can get the correct environment value using bash shell, but using this type shell will lead me to another path problem <a href="https://github.community/t/path-of-bash-shell-in-windows-is-not-correct/17828/2">https://github.community/t5/GitHub-Actions/PATH-of-bash-shell-in-Windows-is-not-correct/m-p/49789/highlight/false#M7607</a>.</p>
